### PR TITLE
Add descriptions to out_file plugin parameters

### PR DESCRIPTION
**Pull request by cosmo0920**
I tried to add descriptions in out_file plugin.

Related to #584.

### CODE REVIEW

File: lib/fluent/plugin/out_file.rb [hunk 1]
```diff
@@ -23,9 +23,13 @@ class FileOutput < TimeSlicedOutput
       'gzip' => :gz,
     }
 
+    desc "Specify filepath to output."
     config_param :path, :string
+    desc "Choose output format. You can choose it from registered formatter plugins."
     config_param :format, :string, :default => 'out_file'
+    desc "Disable path increment."
     config_param :append, :bool, :default => false
+    desc "Compress chunks with specified compression algorithm."
```
Comment (sonots) [inline lib/fluent/plugin/out_file.rb]: Should be

    Compress output file

File: lib/fluent/plugin/out_file.rb [hunk 2]
```diff
@@ -33,6 +37,7 @@ class FileOutput < TimeSlicedOutput
       end
       c
     end
+    desc "A parameter is used for pointing to latest written file."
     config_param :symlink_path, :string, :default => nil
```
Comment (sonots) [inline lib/fluent/plugin/out_file.rb]: Should be like

    Create symlink to temporary buffered file when buffer_type is file.

**Comment by sonots** [comment]
Could you use same words with doc? http://docs.fluentd.org/articles/out_file

File: lib/fluent/plugin/out_file.rb (hunk 1 above)
Comment (cosmo0920) [inline lib/fluent/plugin/out_file.rb]: Document says "Compresses flushed files using gzip.".

Should I use this one or your suggestion?

Comment (sonots) [inline lib/fluent/plugin/out_file.rb]: No. For this, docs should be fixed (not always gzip)

Comment (cosmo0920) [inline lib/fluent/plugin/out_file.rb]: Hmm..., OK I'll fix this line with your suggestion.

Comment (sonots) [inline lib/fluent/plugin/out_file.rb]: Ah, let me write more descriptive. 

Currently, `out_file` supports only gzip, so `Compresses flushed files using gzip` is not wrong, but I think 

    Compress flushed file with specified compression algorithm.

is fine. In addition, documents should list what algorithms can be specified.

**Pull request merged**